### PR TITLE
Add verkle proof verification

### PR DESCRIPTION
**Pull request by matkt**
Adding code to enable verification of the pre state root against a Verkle proof. This code is a wrapper that uses the Rust Verkle library.

### CODE REVIEW

**Comment by matkt** [comment]
@thomas-quadratic if you want to take a look

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/proof/ExecutionWitnessData.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    public Header header;
+    @JsonProperty("transactions")
+    public List<String> transactions;
+    @JsonProperty("withdrawals")
+    public List<String> withdrawals;
+    @JsonProperty("executionWitness")
+    public ExecutionWitness executionWitness;
+
+
+    static class Header {
```
Comment (garyschulte) [inline ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/proof/ExecutionWitnessData.java:21]: I think these nested classes can be record types without the additional serialization annotations, using @JsonSerialize annotation on ExecutionWitnessData.  NBD, just less verbose

File: ipa-multipoint/src/test/resources/invalid_block_72.json [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        "0x599eead7a274c667c6ac8c5c42e9a8a0ffb47e6d2e75818e21ff304e9273f08a",
+        "0x0e029761fcf20de22f3f3e5cfff0b20e7b2510d81422ba70a92608b0672aac21",
+        "0x6d112adf802bf565a2486bbe9511a53df41fad3b643025b4534d0cfc490a3070",
+        "0x1c47dd00f5ecb06a3fa720f1dc767959acf418fdb5ed841b68a4017e426d2864",
+        "0x105381b6bc5ce78b4b8bce6f4461d9669fc298f1ecab2f7ffcca3c82fa7e052e",
+        "0x72e1694b8989f651c3e0b8ffc0673259442cff8aebadb93173a48d78cb658ef1",
+        "0x3ec5a18b934ee02eb8223aea330a5689dc7714b441e29d7dccc0e1e2b02d97ff",
+        "0x6ff5a48312a2d4e85ea7f3daf645cc021abff01ecf68ab414abff1dc35baed4a"
+      ],
+      "d": "0x4cfe43b3414734150b9802c65151e75d5d92fc320ce2140270fc16417b7c323c",
```
Comment (garyschulte) [inline ipa-multipoint/src/test/resources/invalid_block_72.json:175]: least descriptive of all of the struct names 😂. (cry blood unavailable)

**Review by garyschulte — APPROVED** [review]
It looks like we are passing a _lot_ of data.  IDK if it is worth trying to pack this data before we send it, or use native java objects and translate them (like jobjectarray -> vec, etc).

It LGTM in its current form though 👍    Performance optimizations can wait for profiling

**Comment by matkt** [comment]
> It looks like we are passing a _lot_ of data. IDK if it is worth trying to pack this data before we send it, or use native java objects and translate them (like jobjectarray -> vec, etc).

Yes not sure for the moment what is the best. profiling will help to choose the best option

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/proof/VerifyProofTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+public class VerifyProofTest {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    static {
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    @Test
```
[the remark this answers is not in the document]
Comment (matkt) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/proof/VerifyProofTest.java:25]: done

File: ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/proof/ExecutionWitnessData.java (hunk 1 above)
Comment (matkt) [reply ipa-multipoint/src/test/java/org/hyperledger/besu/nativelib/ipa_multipoint/proof/ExecutionWitnessData.java:21]: for the moment the project is still with java 11 from what I see so I cannot use record

**Pull request merged**